From 77b6c0dd6bf174357cb00e37479c64e9eaa70064 Mon Sep 17 00:00:00 2001
From: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Date: Fri, 20 Oct 2023 13:35:41 +0200
Subject: [PATCH 0593/1141] clk: stm32mp25: remove ck_icn_p_gicv2m clock

ck_icn_p_gicv2m is considered as system clocks and then managed by OPTEE.

Signed-off-by: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Change-Id: I61e7bac982e37708f840a150b80f0354267fabf5
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/335064
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
Reviewed-by: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
---
 drivers/clk/stm32/clk-stm32mp25.c | 8 ++------
 1 file changed, 2 insertions(+), 6 deletions(-)

diff --git a/drivers/clk/stm32/clk-stm32mp25.c b/drivers/clk/stm32/clk-stm32mp25.c
index 268dbce0b678..024e746e4448 100644
--- a/drivers/clk/stm32/clk-stm32mp25.c
+++ b/drivers/clk/stm32/clk-stm32mp25.c
@@ -286,7 +286,6 @@ enum enum_gate_cfg {
 	GATE_CRC,
 	GATE_SERC,
 	GATE_OSPIIOM,
-	GATE_GICV2M,
 	GATE_I3C1,
 	GATE_I3C2,
 	GATE_I3C3,
@@ -358,6 +357,7 @@ enum enum_gate_cfg {
 	GATE_RTC,
 	GATE_SYSCPU1,
 	GATE_BSEC,
+	GATE_GICV2M,
 #endif
 	GATE_NB
 };
@@ -490,7 +490,6 @@ static const struct stm32_gate_cfg stm32mp25_gates[GATE_NB] = {
 	GATE_CFG(GATE_CRC,		RCC_CRCCFGR,		1,	0),
 	GATE_CFG(GATE_SERC,		RCC_SERCCFGR,		1,	0),
 	GATE_CFG(GATE_OSPIIOM,		RCC_OSPIIOMCFGR,	1,	0),
-	GATE_CFG(GATE_GICV2M,		RCC_GICV2MCFGR,		1,	0),
 	GATE_CFG(GATE_I3C1,		RCC_I3C1CFGR,		1,	0),
 	GATE_CFG(GATE_I3C2,		RCC_I3C2CFGR,		1,	0),
 	GATE_CFG(GATE_I3C3,		RCC_I3C3CFGR,		1,	0),
@@ -562,6 +561,7 @@ static const struct stm32_gate_cfg stm32mp25_gates[GATE_NB] = {
 	GATE_CFG(GATE_RTC,		RCC_RTCCFGR,		1,	0),
 	GATE_CFG(GATE_SYSCPU1,		RCC_SYSCPU1CFGR,	1,	0),
 	GATE_CFG(GATE_BSEC,		RCC_BSECCFGR,		1,	0),
+	GATE_CFG(GATE_GICV2M,		RCC_GICV2MCFGR,		1,	0),
 #endif
 };
 
@@ -708,9 +708,6 @@ static CLK_STM32_GATE(ck_icn_p_ethsw_acm_msg, "ck_icn_ls_mcu", 0, GATE_ETHSWACMM
 static CLK_STM32_GATE(ck_icn_p_fdcan, "ck_icn_apb2", 0, GATE_FDCAN);
 static CLK_STM32_GATE(ck_ker_fdcan, "ck_flexgen_26", 0, GATE_FDCAN);
 
-/* GICV2M */
-static CLK_STM32_GATE(ck_icn_p_gicv2m, "ck_icn_apb4", CLK_IS_CRITICAL, GATE_GICV2M);
-
 /* GPU */
 static CLK_STM32_GATE(ck_icn_m_gpu, "ck_flexgen_59", 0, GATE_GPU);
 static CLK_STM32_GATE(ck_ker_gpu, "ck_pll3", 0, GATE_GPU);
@@ -1094,7 +1091,6 @@ static const struct clock_config stm32mp25_clock_cfg[] = {
 	STM32_GATE_CFG(CK_BUS_CSI, ck_icn_p_csi, SEC_RIFSC(CSI)),
 	STM32_GATE_CFG(CK_BUS_DCMIPP, ck_icn_p_dcmipp, SEC_RIFSC(DCMIPP)),
 	STM32_GATE_CFG(CK_BUS_LVDS, ck_icn_p_lvds, SEC_RIFSC(LVDS)),
-	STM32_GATE_CFG(CK_BUS_GICV2M, ck_icn_p_gicv2m, SEC_RIFSC(GICV2M)),
 	STM32_GATE_CFG(CK_BUS_USBTC, ck_icn_p_usbtc, SEC_RIFSC(UCPD1)),
 	STM32_GATE_CFG(CK_BUS_USB3PCIEPHY, ck_icn_p_usb3pciephy, SEC_RIFSC(USB3DR)),
 	STM32_GATE_CFG(CK_BUS_VDEC, ck_icn_p_vdec, SEC_RIFSC(VDEC)),
-- 
2.39.2

